source/03-components/table/table--sortable.es6.ts

```diff
@@ -0,0 +1,295 @@
+import Drupal from 'drupal';
+import once from 'once';
+
+const TABLE = `.c-table.is-sortable`;
+const SORTED = 'aria-sort';
+const ASCENDING = 'ascending';
+const DESCENDING = 'descending';
+const SORT_OVERRIDE = 'data-sort-value';
+const SORT_BUTTON_CLASS = `c-table__header-button`;
+const SORT_BUTTON = `.${SORT_BUTTON_CLASS}`;
+const SORTABLE_ATTRIBUTE = `data-sortable`;
+const SORTABLE_HEADER = `th[${SORTABLE_ATTRIBUTE}]`;
+const ANNOUNCEMENT_CLASS = `c-table__announcement-region`;
+const ANNOUNCEMENT_REGION = `.${ANNOUNCEMENT_CLASS}[aria-live="polite"]`;
+const SORT_ALL_COLUMNS = true;
+
+/**
+ * Gets the data-sort-value attribute value, if provided — otherwise, gets
+ * the innerText or textContent of the child element (HTMLTableCellElement)
+ * at the specified index of the given table row.
+ *
+ * @param {number} index
+ * @param {HTMLTableRowElement} tr
+ * @return {string | number | null}
+ */
+const getCellValue = (
+  tr: HTMLTableRowElement,
+  index: number
+): string | number | null =>
+  (tr.children[index] as HTMLTableCellElement).getAttribute(SORT_OVERRIDE) ||
+  (tr.children[index] as HTMLTableCellElement).innerText ||
+  (tr.children[index] as HTMLTableCellElement).textContent;
```

Comment:
Should we be casting like this instead of doing some sort of check? For instance, we could be doing something like the following for better type safety:
```ts
  const item = tr.children[index];
  if (item instanceof HTMLTableCellElement) {
    return (
      item.getAttribute(SORT_OVERRIDE) || item.innerText || item.textContent
    );
  }
  // Item not HTMLTableCellElement.
  return null;
```